flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -608,6 +610,33 @@ public String getValue(ConfigOption<?> configOption) {
 		return o == null ? null : o.toString();
 	}
 
+	/**
+	 * Returns the value associated with the given config option as an enum.
+	 *
+	 * @param enumClass    The return enum class
+	 * @param configOption The configuration option
+	 * @throws IllegalArgumentException If the string associated with the given config option cannot
+	 *                                  be parsed as a value of the provided enum class.
+	 */
+	@PublicEvolving
+	public <T extends Enum<T>> T getAsEnum(
+			final Class<T> enumClass,
+			final ConfigOption<String> configOption) {
+		checkNotNull(enumClass, "enumClass must not be null");
+		checkNotNull(configOption, "configOption must not be null");
+
+		final String configValue = getString(configOption);
+		try {
+			return T.valueOf(enumClass, configValue);
```

Comment:
Shouldn't we call `toUpperCase` on `configValue`? At least for the scope of `yarn.per-job-cluster.include-user-jar` we tried to be case insensitive so far, which seems fair.

Reply:
Behavior shouldn't change. Fixed it.

Reply:
You are right. I added a test case for this.